api/envoy/api/v2/route/route_components.proto

```diff
@@ -1125,13 +1125,13 @@ message HedgePolicy {
   // [#not-implemented-hide:]
   type.FractionalPercent additional_request_chance = 2;
 
-  // Indicates that a hedged request should be sent when the per-try timeout
-  // is hit. This will only occur if the retry policy also indicates that a
-  // timed out request should be retried.
-  // Once a timed out request is retried due to per try timeout, the router
-  // filter will ensure that it is not retried again even if the returned
-  // response headers would otherwise be retried according the specified
-  // :ref:`RetryPolicy <envoy_api_msg_route.RetryPolicy>`.
+  // Indicates that a hedged request should be sent when the per-try timeout is hit.
+  // This will only occur if the :ref:`RetryPolicy <envoy_api_msg_route.RetryPolicy>` also indicates that
+  // timed out requests should be retried (e.g. retry_on set to 'gateway-error' etc). (Other retry policies
+  // would also apply, but would only have effect if the response came back before the request was hedged against;
+  // otherwise such responses would simply be discarded as a retry is already in flight.)
```

Comment:
I think this might be clearer stated as

"Any response received after the timeout and subsequent hedge attempt will never be retried, no matter the RetryPolicy"

Reply:
Assuming timeout 150ms, per-try timeout of 50ms, 3 retries and retry-on: 5xx policy, and hedging enabled:

0ms: Request 1 sent.
50ms: Request 1 times out, (hedged) request 2 sent.
75ms: Request 2 (hedged) returns 500.
150ms: Request 1 times out.

Would there be a 3rd request?

Reply:
there would be a 3rd request. Request 2 is considered a new attempt, so it will be retried if it times out or returns a 500. If request 1 comes back with a 500 after request 2 has already been sent, that will be dropped and not retried

Reply:
ahh, I read `(Any response received after the timeout) AND (subsequent hedge attempt) -> will never be retried` 🤦